demo/kotlin/src/main/kotlin/RiseSetCulm.kt

```diff
@@ -34,19 +34,13 @@ internal fun `Rise Set Culmination demo`(observer: Observer, startTime: Time): I
     )
 
     // Print the list in sorted order.
-    eventList.sorted().forEach {
-        it.display()
-    }
+    eventList.sorted().forEach(AstroEvent::display)
 
     return 0
 }
 
 
-private fun maybe(name: String, time: Time?): AstroEvent? =
-    if (time == null)
-        null
-    else
-        AstroEvent(name, time)
+private fun maybe(name: String, time: Time?): AstroEvent? = time?.let { AstroEvent(name, it) }
```

Comment:
This `.let {}` in Kotlin is similar to `map`/`Select` except applies to single values, when used in `?.let` it maps the value of nullable to another nullable value and it is `inline` so no much need to worry performance wise on other cases you may like to use,

```
@kotlin.internal.InlineOnly
public inline fun <T, R> T.let(block: (T) -> R): R {
    contract {
        callsInPlace(block, InvocationKind.EXACTLY_ONCE)
    }
    return block(this)
}
```

Reply:
That is great to know. And in demos, I'm not worried as much about performance as making things obvious. If something is more confusing in a demo, I will leave it out. But this is pretty clear and makes the code smaller, so I like it!